markets/storageadapter/provider.go

```diff
@@ -57,10 +58,12 @@ type ProviderNodeAdapter struct {
 	maxDealCollateralMultiplier uint64
 	dsMatcher                   *dealStateMatcher
 	scMgr                       *SectorCommittedManager
+
+	rpn retrievalmarket.RetrievalProviderNode
 }
 
-func NewProviderNodeAdapter(fc *config.MinerFeeConfig, dc *config.DealmakingConfig) func(mctx helpers.MetricsCtx, lc fx.Lifecycle, dag dtypes.StagingDAG, secb *sectorblocks.SectorBlocks, full v1api.FullNode, dealPublisher *DealPublisher) storagemarket.StorageProviderNode {
-	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, dag dtypes.StagingDAG, secb *sectorblocks.SectorBlocks, full v1api.FullNode, dealPublisher *DealPublisher) storagemarket.StorageProviderNode {
+func NewProviderNodeAdapter(fc *config.MinerFeeConfig, dc *config.DealmakingConfig) func(mctx helpers.MetricsCtx, lc fx.Lifecycle, dag dtypes.StagingDAG, secb *sectorblocks.SectorBlocks, full v1api.FullNode, dealPublisher *DealPublisher, rpn retrievalmarket.RetrievalProviderNode) storagemarket.StorageProviderNode {
```

Comment:
So, as best I understand, we have the following:

Intention: RetrievalProviderNode is intended to be a facade interface for the retrieval provider code in go-fil-markets to talk to either a markets process or miner process for functionality not provided in the markets library. It's just an interface definition. It should include only methods used by retrieval code in go-fil-markets.

Currently: RetrievalProviderNode provides two functions:
IsUnsealed
UnsealSector

UnsealSector is not used by retrieval market code, and thus, should not be part of RetrievalProviderNode.

UnsealSector is however required by markets/dagstore/MinerAPI

IsUnsealed is used by markets/dagstore/MinerAPI AND by the implementation of StorageProviderNode -- but IsUnsealed is not actually used on the StorageProviderNode in the markets implementation -- it's not clear why it's added.

Both MinerAPI and StorageProviderNode depend on RetrievalProviderNode.... that seems super awkward.

It seems like we have some untangling to do.

First, RetrievalProviderNode, and StorageMarketNode are just interfaces -- they're defined at the site of use, in go-fil-markets, per Go lang best practices (see https://github.com/golang/go/wiki/CodeReviewComments#interfaces). There's no need to get attached to that collection of methods as having any inherent meaning for code in this repo, other than you have to put together something that implements everything requires by markets code in order to initialize a storage provider / retrieval provider. 

My recommendations are:
- Remove UnsealSector from RetrievalProviderNode
- Remove IsUnsealed from StorageProviderNode

- From markets/retrievaladapter/provider, extract IsUnsealed and UnsealSector implementations into a common dependency that can embed in MinerAPI & markets/retrievaladapter/provider (you can just have it be an embedded struct to avoid the one line methods)
- You can just use this in StorageProviderNode if you need to add IsUnsealed at some later point.

Reply:
Agree with all of this 👍

Reply:
Implemented the change in
- [x] markets https://github.com/filecoin-project/go-fil-markets/pull/601
- [ ] lotus